src/main/java/ch/njol/skript/expressions/ExprAffectedEntities.java

```diff
@@ -58,20 +61,52 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 		return true;
 	}
 
-	@Nullable
 	@Override
-	protected LivingEntity[] get(Event e) {
-		if (e instanceof AreaEffectCloudApplyEvent)
-			return ((AreaEffectCloudApplyEvent) e).getAffectedEntities().toArray(new LivingEntity[0]);
+	protected LivingEntity @Nullable [] get(Event event) {
+		if (event instanceof AreaEffectCloudApplyEvent areaEvent)
+			return areaEvent.getAffectedEntities().toArray(new LivingEntity[0]);
 		return null;
 	}
 
-	@Nullable
 	@Override
-	public Iterator<? extends LivingEntity> iterator(Event e) {
-		if (e instanceof AreaEffectCloudApplyEvent)
-			return ((AreaEffectCloudApplyEvent) e).getAffectedEntities().iterator();
-		return super.iterator(e);
+	public @Nullable Iterator<? extends LivingEntity> iterator(Event event) {
+		if (event instanceof AreaEffectCloudApplyEvent areaEvent)
+			return areaEvent.getAffectedEntities().iterator();
+		return super.iterator(event);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case ADD, SET, DELETE, REMOVE -> CollectionUtils.array(LivingEntity[].class);
+			default -> null;
+		};
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		if (!(event instanceof AreaEffectCloudApplyEvent areaEvent))
+			return;
+
+		LivingEntity[] entities = (LivingEntity[]) delta;
+		switch (mode) {
+			case REMOVE:
```

Comment:
use enhanced switch

Reply:
I would be removing the fall through if I switched to an enhanced switch. Which I assume is what Sovde wants considering one of his previous requested changes

Reply:
You can use one if you want, you just have to make sure the behavior is the same
Keeping it as is is probably better though